dsl_parser/elements/data_types.py

```diff
@@ -0,0 +1,121 @@
+########
+# Copyright (c) 2013 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+from dsl_parser import constants
+from dsl_parser import exceptions
+from dsl_parser import utils
+from dsl_parser.elements import properties_utils
+from dsl_parser.framework.elements import (
+    Element,
+    Dict,
+    DictElement,
+    StringElement)
+from dsl_parser.framework.parser import parse
+from dsl_parser.framework.requirements import Value
+
+
+_DERIVED_FROM = 'derived_from'
+_TYPE = 'type'
+
+
+class DataType(Element):
+    schema = {
+        constants.PROPERTIES: properties_utils.UnsafeSchema,
+        'description': StringElement,
+        _DERIVED_FROM: StringElement,
+        'version': StringElement
+    }
+
+    def parse(self):
+        result = self.build_dict_result()
+        if constants.PROPERTIES not in result:
+            result[constants.PROPERTIES] = {}
+        return result
+
+
+def _add_requirement_if_exists(current_type,
+                               required_type,
+                               types_classes,
+                               unknown_type_err):
+    if required_type not in types_classes:
+        raise exceptions.DSLParsingElementMatchException(
+            39,
```

Comment:
Verify error code is not used anywhere else and move as constant to dsl_parser.exceptions